core/input/input.cpp

```diff
@@ -870,6 +870,25 @@ void Input::parse_input_event(const Ref<InputEvent> &p_event) {
 
 	ERR_FAIL_COND(p_event.is_null());
 
+#ifdef DEBUG_ENABLED
+	uint64_t curr_frame = Engine::get_singleton()->get_process_frames();
+	if (p_event->_get_parsed_frame() == curr_frame) {
+		// It would be technically safe to send the same event in cases such as:
+		// - After an explicit flush.
+		// - In platforms using buffering when agile flushing is enabled, after one of the mid-frame flushes.
+		// - If platform doesn't use buffering and event accumulation is disabled.
+		// - If platform doesn't use buffering and the event type is not accumulable.
+		// However, it wouldn't be reasonable to ask users to remember the full ruleset and be aware at all times of the possibilites of the
+		// target platform, project settings and engine internals, which may change without notice.
+		// Therefore, the guideline is, "don't send the same event object more than once per frame".
+		WARN_PRINT(
```

Comment:
Maybe `WARN_PRINT_ONCE` as I assume this can easily spam?

Reply:
Reasonable.